File: RealLiteUi.TypeScript/html/componente/login/DivLoginEntrar.ts
/// <reference path="../../../../RealLifeShared.TypeScript/dominio/ContaDominio.ts"/>
/// <reference path="../../../../Web.TypeScript/Utils.ts"/>

module RealLifeUi
{
    // #region Importações

    import ContaDominio = RealLifeShared.ContaDominio;
    import Utils = Web.Utils;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class DivLoginEntrar extends DivLoginBase
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        private _cmpStrLogin: CampoRealLifeUi;
        private _cmpStrSenha: CampoRealLifeUi;

        private get cmpStrLogin(): CampoRealLifeUi
        {
            if (this._cmpStrLogin != null)
            {
                return this._cmpStrLogin;
            }

            this._cmpStrLogin = new CampoRealLifeUi(this.strId + "_cmpStrLogin");

            return this._cmpStrLogin;
        }

        private get cmpStrSenha(): CampoRealLifeUi
        {
            if (this._cmpStrSenha != null)
            {
                return this._cmpStrSenha;
            }

            this._cmpStrSenha = new CampoRealLifeUi(this.strId + "_cmpStrSenha");

            return this._cmpStrSenha;
        }

        // #endregion Atributos

        // #region Construtores

        constructor()
        {
            super(DivLoginEntrar.name);
        }

        // #endregion Construtores

        // #region Métodos

        public carregarDados(objConta: ContaDominio): void
        {
            if (objConta == null)
            {
                return;
            }

            if (this.cmpStrLogin.txt.strValor.indexOf("@") > -1)
            {
                objConta.strEmail = this.cmpStrLogin.txt.strValor;
            }
            else
            {
                objConta.strGametag = this.cmpStrLogin.txt.strValor;
            }

            objConta.strSenha = this.encriptar(this.cmpStrSenha.txt.strValor);
        }

        public entrarValidar(): boolean
        {
            if (Utils.getBooStrVazia(this.cmpStrLogin.txt.strValor))
            {
                this.cmpStrLogin.txt.receberFoco();
                return false;
            }

            if (Utils.getBooStrVazia(this.cmpStrSenha.txt.strValor))
            {
                this.cmpStrSenha.txt.receberFoco();
                return false;
            }

            return true;
        }

        protected inicializar(): void
        {
            super.inicializar();

            this.cmpStrLogin.iniciar();
            this.cmpStrSenha.iniciar();

            window.setTimeout((() => { this.cmpStrLogin.txt.receberFoco(); }), 350);
        }

        // #endregion Métodos

        // #region Eventos
        // #endregion Eventos
    }
}

File: RealLifeUi/Html/Componente/Login/DivLoginEntrar.cs
using NetZ.Web.Html.Componente.Botao;

namespace RealLifeUi.Html.Componente.Login
{
    internal class DivLoginEntrar : DivLoginBase
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        private CampoRealLifeUi _cmpStrLogin;
        private CampoRealLifeUi _cmpStrSenha;

        private CampoRealLifeUi cmpStrLogin
        {
            get
            {
                if (_cmpStrLogin != null)
                {
                    return _cmpStrLogin;
                }

                _cmpStrLogin = new CampoRealLifeUi("User");

                return _cmpStrLogin;
            }
        }

        private CampoRealLifeUi cmpStrSenha
        {
            get
            {
                if (_cmpStrSenha != null)
                {
                    return _cmpStrSenha;
                }

                _cmpStrSenha = new CampoRealLifeUi("Senha");

                return _cmpStrSenha;
            }
        }

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        protected override void inicializar()
        {
            base.inicializar();

            this.cmpStrSenha.booUltimo = true;
            this.cmpStrSenha.txt.enmTipo = NetZ.Web.Html.Input.EnmTipo.PASSWORD;
        }

        protected override void montarLayout()
        {
            base.montarLayout();

            this.cmpStrLogin.setPai(this.divConteudo);
            this.cmpStrSenha.setPai(this.divConteudo);
        }

        protected override void setStrId(string strId)
        {
            base.setStrId(strId);

            this.cmpStrLogin.strId = (this.strId + "_cmpStrLogin");
            this.cmpStrSenha.strId = (this.strId + "_cmpStrSenha");
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}